bindings/redis/metadata.yaml

```diff
@@ -0,0 +1,207 @@
+    default: "node"
+    description: |
+      Redis service type. Set to "node" for single-node mode, or "cluster" for Redis Cluster.
+    example: "cluster"
+  - name: redisMinRetryInterval
+    required: false
+    description: |
+      Minimum backoff for redis commands between each retry. "-1" disables backoff.
+    default: "8ms"
+    example: "-1"
+    type: duration
+  - name: dialTimeout
+    required: false
+    description: Dial timeout for establishing new connections.
+    default: "5s"
+    example: "10s"
+    type: duration
+  - name: readTimeout
+    required: false
+    description: |
+      Timeout for socket reads. If reached, redis commands will fail with a
+      timeout instead of blocking. Use "-1" for no timeout.
+    default: "3s"
+    example: "10s"
+    type: duration
+  - name: writeTimeout
+    required: false
+    description: |
+      Timeout for socket writes. If reached, redis commands will fail with
+      a timeout instead of blocking. Defaults is "readTimeout".
+    example: "3s"
+    type: duration
+  - name: poolSize
+    required: false
+    description: |
+      Maximum number of socket connections. Default is 10 connections per
+      every CPU as reported by runtime.NumCPU.
+    example: "20"
+    type: number
+  - name: poolTimeout
+    required: false
+    description: |
+      Amount of time client waits for a connection if all connections are busy
+      before returning an error. Default is readTimeout + 1 second.
+    example: "5s"
+    type: duration
+  - name: maxConnAge
+    required: false
+    description: |
+      Connection age at which the client retires (closes) the connection.
+      Default is to not close aged connections.
+    example: "30m"
+    type: duration
+  - name: minIdleConns
+    required: false
+    description: |
+      Minimum number of idle connections to keep open in order to avoid
+      the performance degradation associated with creating new connections.
+    default: "0"
+    example: "2"
+    type: number
+  - name: idleCheckFrequency
+    required: false
+    description: |
+      Frequency of idle checks made by idle connections reaper.
+      "-1" disables idle connections reaper.
+    default: "1m"
+    example: "-1"
+    type: duration
+  - name: idleTimeout
+    required: false
+    description: |
+      Amount of time after which the client closes idle connections. Should be
+      less than server's timeout.
+      "-1" disables idle timeout check.
+    default: "5m"
+    example: "10m"
+    type: duration
+  - name: ttlInSeconds
+    required: false
+    description: |
+      Allows specifying a default Time-to-live (TTL) in seconds that will be
+      applied to every state store request unless TTL is explicitly defined
+      via the request metadata.
+    example: "600"
+    type: number
+  - name: queryIndexes
+    required: false
+    description: Indexing schemas for querying JSON objects
+    example: "see Querying JSON objects"
+    type: string
```

Comment:
If I comment all this out and run `make check-component-metadata && make bundle-component-metadata` the command only complains about the following properties:

```
The following components are missing metadata in their metadata.yaml:
{
  "bindings/redis": {
    "maxretries": "int",
    "maxretrybackoff": "time.Duration",
    "queryindexes": "string",
    "ttlinseconds": "*int"
  }
}
```

Like 🤯. What is happening here?

Reply:
But, these are optional attributes, as far as I understand. Am I missing something?

Reply:
But, all these mentioned 4 metadata are optional. Am I missing something?

Reply:
If I only define those 4 metadata and nothing else that is in this metadata.yaml file, `make check-component-data` would pass.

What is puzzling me is that these fields (or a structure hosting only these fields) are not defined in any of the structs used for metadata and settings by bindings/redis. 

> Am I missing something?

That is exactly my question. I am even wondering: am I missing something really obvious and should I instead define only these 4 fields `make check-component-data`  is requiring instead ?

Reply:
ALrigh! We found the issue and it is addressed on #2803.

Pending its merge before procedding. Moving this back to draft.